docs/JsonSchema.md

```diff
@@ -0,0 +1,331 @@
+
+### MovableList
+
+#### Insert
+
+```ts
+{
+    "type": "insert",
+    "pos": number,
+    "value": LoroValue
+}
+```
+
+- `type`: `insert`, `delete`, `move` or `set`.
+- `pos`: the index of the insert operation.
+- `value`: the insert content which is a list of `LoroValue`
+
+#### Delete
+
+```ts
+{
+    "type": "delete",
+    "pos": number,
+    "len": number,
+    "delete_start_id": string
+}
+```
+
+- `type`:`insert`, `delete`, `move` or `set`.
+- `pos`: the start index of the deletion.
+- `len`: the length of deleted content.
+- `delete_start_id`: the string id of start element.
+
+#### Move
+
+```ts
+{
+    "type": "move",
+    "from": number,
+    "to": number,
+    "from_id": string
+}
+```
+
+- `type`:`insert`, `delete`, `move` or `set`.
+- `from`: the index of the element before is moved.
+- `to`: the index of the index moved to after moving out the element
+- `from_id`: the string id of the element moved.
+
+#### Set
+
+```ts
+{
+    "type": "set",
+    "elem_id": string,
+    "value": LoroValue
+}
+```
+
+- `type`:`insert`, `delete`, `move` or `set`.
+- `elem_id`: the string id of the element replaced.
+- `value`: the value setted.
+
+### Map
+
+#### Insert
+
+```ts
+{
+    "type": "insert",
+    "key": string,
+    "value": LoroValue
+}
+```
+
+- `type`: `insert` or `delete`.
+- `key`: the key of the insertion.
+- `value`: the value of the insertion.
+
+#### Delete
+
+```ts
+{
+    "type": "delete",
+    "key": string
+}
+```
+
+- `type`: `insert` or `delete`.
+- `key`: the key of the deletion
+
+### Text
+
+#### Insert
+
+```ts
+{
+    "type": "insert",
+    "pos": number,
+    "text": string
+}
+```
+
+#### Delete
+
+```ts
+{
+    "type": "delete",
+    "pos": number,
+    "len": number,
+    "id_start": string
+}
+```
+
+#### Mark
+
+```ts
+{
+    "type": "mark",
+    "start": number,
+    "end": number,
+    "style_key": string,
+    "style_value": LoroValue,
+    "info": number
+}
+```
+
+#### MarkEnd
+
+```ts
+{
+    "type": "mark_end"
+}
+```
+
+### Tree
+
+#### Move
+
+```ts
+{
+    "type": "move",
+    "target": string,
+    "parent": string | null,
+    "fractional_index": UInt8Array
+}
+```
+
+- `type`: `move` or `delete`.
+- `target`: the string format of target `TreeID` moved.
+- `parent`: the string format of `TreeID` or `null`. If it is `null`, the target node will be a root node.
+- `fractional_index`: the fractional index of the target node.
+
+#### Delete
+
+```ts
+{
+    "type": "delete",
+    "target": string
+}
+```
+
+- `type`: `move` or `delete`.
+- `target`: the string format of target `TreeID` deleted.
+
+### Unknown
+
+To support backward and forward compatibility of encoding, we have an unknonw type. When an `Op` from newer version is decoded into the older version, it will be treated as an unknown type which is in a more general form, such as binary and string. Conversely, when the new version decodes an unknown `Op`, Loro will know its true type and perform the proper decoding process.
```

Comment:
```suggestion
To support backward and forward compatibility of encoding, we have an unknown type. When an `Op` with a newly supported Container from a newer version is decoded into the older version, it will be treated as an unknown type in a more general form, such as binary and string. When the new version decodes an unknown `Op`, the newer version of Loro will know its true type and decode correctly.
```